app/assets/javascripts/controllers/container_live_dashboard/container_live_dashboard_controller.js

```diff
@@ -0,0 +1,287 @@
+    var pathname = $window.location.pathname.replace(/\/$/, '');
+    var id = '/' + (/^\/[^\/]+\/(\d+)$/.exec(pathname)[1]);
+    var url = '/container_dashboard/data' + id + '/?live=true&tenant=' + tenant;
+
+    var filterChange = function (filters) {
+      $scope.filterChanged = true;
+      $scope.filtersText = "";
+      $scope.tags = {};
+      $scope.filterConfig.appliedFilters.forEach(function (filter) {
+        $scope.filtersText += filter.title + " : " + filter.value + "\n";
+        $scope.tags[filter.id] = filter.value;
+      });
+    };
+
+    $scope.filterConfig = {
+      fields: [],
+      resultsCount: $scope.items.length,
+      appliedFilters: [],
+      onFilterChange: filterChange
+    };
+
+    var selectionChange = function() {
+      $scope.itemSelected = false;
+      for (var i = 0; i < $scope.items.length && !$scope.itemSelected; i++) {
+        if ($scope.items[i].selected) {
+          $scope.itemSelected = true;
+        }
+      }
+    };
+
+    $scope.doApply = function() {
+      $scope.applied = true;
+      $scope.filterChanged = false;
+      $scope.refresh();
+    };
+
+    $scope.doViewGraph = function() {
+      $scope.viewGraph = true;
+      $scope.chartDataInit = false;
+      $scope.refresh_graph_data();
+    };
+
+    $scope.doViewMetrics = function() {
+      $scope.viewGraph = false;
+      $scope.refresh();
+    };
+
+    $scope.actionsConfig = {
+      actionsInclude: true
+    };
+
+    $scope.toolbarConfig = {
+      filterConfig: $scope.filterConfig,
+      actionsConfig: $scope.actionsConfig
+    };
+
+    $scope.graphToolbarConfig = {
+      actionsConfig: $scope.actionsConfig
+    };
+
+    $scope.itemSelected = false;
+
+    $scope.listConfig = {
+      selectionMatchProp: 'id',
+      showSelectBox: true,
+      useExpandingRows: true,
+      onCheckBoxChange: selectionChange
+    };
+
+    var getMetricTags = function() {
+      $http.get(url + '&query=metric_tags').success(function(response) {
+        $scope.tagsLoaded = true;
+        if (response && angular.isArray(response.metric_tags)) {
+          response.metric_tags.sort();
+          for (var i = 0; i < response.metric_tags.length; i++) {
+            $scope.filterConfig.fields.push(
+              {
+                id: response.metric_tags[i],
+                title:  response.metric_tags[i],
+                placeholder: sprintf(__("Filter by %s..."), response.metric_tags[i]),
+                filterType: 'alpha'
+              });
+          }
+        } else {
+          // No filters available, apply without filtering
+          $scope.toolbarConfig.filterConfig = undefined;
+          $scope.doApply();
+        }
+      });
+    };
+
+    var getLatestData = function(item) {
+      var params = '&query=get_data&metric_id=' + item.id + '&limit=5&order=DESC';
+
+      $http.get(url + params).success(function (response) {
+        'use strict';
+        if (response.error) {
+          // TODO: do something ?
```

Comment:
You can probably use `add_flash` to add a flash notification about the failure (you can even give it a unique id so that it only adds it if not already displayed..)